packages/rocketchat-cloud/server/methods.js

```diff
@@ -0,0 +1,131 @@
+		}
+
+		if (!RocketChat.authz.hasPermission(Meteor.userId(), 'manage-cloud')) {
+			throw new Meteor.Error('error-not-authorized', 'Not authorized', { method: 'cloud:retrieveRegistrationInfo' });
+		}
+
+		const info = {
+			canConnect: RocketChat.settings.get('Register_Server'),
+			token: '',
+			email: '',
+		};
+
+		const firstUser = RocketChat.models.Users.getOldest({ emails: 1 });
+		info.email = firstUser && firstUser.emails[0].address;
+
+		if (RocketChat.models.Settings.findOne('Organization_Email')) {
+			info.email = RocketChat.settings.get('Organization_Email');
+		}
+
+		return info;
+	},
+	'cloud:updateEmail'(email) {
+		check(email, String);
+
+		if (!Meteor.userId()) {
+			throw new Meteor.Error('error-invalid-user', 'Invalid user', { method: 'cloud:updateEmail' });
+		}
+
+		if (!RocketChat.authz.hasPermission(Meteor.userId(), 'manage-cloud')) {
+			throw new Meteor.Error('error-not-authorized', 'Not authorized', { method: 'cloud:updateEmail' });
+		}
+
+		RocketChat.models.Settings.updateValueById('Organization_Email', email);
+	},
+	'cloud:connectServer'(token) {
+		check(token, String);
+
+		if (!Meteor.userId()) {
+			throw new Meteor.Error('error-invalid-user', 'Invalid user', { method: 'cloud:connectServer' });
+		}
+
+		if (!RocketChat.authz.hasPermission(Meteor.userId(), 'manage-cloud')) {
+			throw new Meteor.Error('error-not-authorized', 'Not authorized', { method: 'cloud:connectServer' });
+		}
+
+		const redirectUrl = `${ RocketChat.settings.get('Site_Url') }/api/cloud/redirect`.replace(/\/\/+/g, '/');
+
+		const regInfo = {
+			email: RocketChat.settings.get('Organization_Email'),
+			client_name: RocketChat.settings.get('Site_Name'),
+			redirectUris: [redirectUrl],
+		};
+
+		console.log(regInfo);
+
+		const cloudUrl = RocketChat.settings.get('Cloud_Url');
+		const result = HTTP.post(`${ cloudUrl }/api/oauth/clients`, {
+			headers: {
+				Authorization: `Bearer ${ token }`,
+			},
+			data: regInfo,
+		});
+
+		const { data } = result;
+
+		const state = Random.id();
+		RocketChat.models.Settings.updateValueById('Cloud_Workspace_Id', data.workspaceId);
+		RocketChat.models.Settings.updateValueById('Cloud_Workspace_Name', data.client_name);
+		RocketChat.models.Settings.updateValueById('Cloud_Workspace_Client_Id', data.client_id);
+		RocketChat.models.Settings.updateValueById('Cloud_Workspace_Client_Secret', data.client_secret);
+		RocketChat.models.Settings.updateValueById('Cloud_Workspace_Client_Secret_Expires_At', data.client_secret_expires_at);
+		RocketChat.models.Settings.updateValueById('Cloud_Workspace_Registration_Client_Uri', data.registration_client_uri);
+		RocketChat.models.Settings.updateValueById('Cloud_Workspace_Registration_State', state);
+
+		const url = `${ cloudUrl }/authorize?response_type=code&client_id=${ data.client_id }&redirect_uri=${ redirectUrl }&scope=offline+workspace&state=${ state }`;
+
+		console.log('User is being sent to:', url);
```

Comment:
if this log is relevant, please consider changing to `import { Logger } from 'meteor/rocketchat:logger';`